### DISCUSSION TITLE
Initial conditions for phase field simulations

### DISCUSSION THREAD

**Discussion by Giovar6** [idaholab/moose#19333] [Q&A Modules: Phase field] [answered]
Dear MOOSE developers,

I would like to ask for some information about the initial conditions. 
I am currently simulating the sintering of some powder particles. At present day I have found that a possible way to specify the initial conditions is to use the SmoothCircleIC to specify the initial conditions for the eta parameter of each particle and then the SpecifiedSmoothCircleIC to specify in one function all the initial conditions for the "c" field. 
So, now I have two main questions:
1) If I would like to include in the same model some circular powder particles and a box, how I can specify the initial conditions for the "c" field? Is there a possibility to combine the SmoothCirlceIC with the BoundingBoxIC, just as an example?
2) If I would like to specify the initial conditions for powder particles that do not have a regular shape, what type of function should I use? Should I have to write the code from scratch?
Thank you in advance to anyone that will help me.

**Comment by dschwen** [comment, marked as answer]
1. You can use function ICs or `CoupledValueFunctionIC`. The latter can couple variables that are initialized by other ICs.
2. Writing from scratch is always an option. But take a look at `ImageFunction`, which can load images that you can prepare with any paint program. Together with (again) `FunctionIC` you can use this to "draw" your initial condition, which seems suitable for irregular particle ICs.

**Reply by Giovar6** [reply]
Thank you very much for your precious help.
As regards question number 1 I  have followed your suggestion and I have correctly implemented the two different initial conditions.
As regards question number 2, I am now facing an issue.
I would like to specify a threshold, in order to apply a certain value outside and inside the particles. When I do this operation, though I have an 8-bit black and white image with a threshold of 255 I have some problems.
If I use the following code for the function:
 ``` 
[ICFromImagec]
    type = ImageFunction
    file = Cbedt.png
    threshold = 255
    upper_value = 0.999
    lower_value = 1e-10
  [../]
````
The simulation converge but I obtain ICS opposite to what I was expecting, with 1 outside the material and 0 inside.
If I switch upper and lower values, the simulation does not converge. 
If I use as a upper_value = 0.001 and as a lower_value = 0.999, the simulation converges and the ICS are set properly, 
but, strangely, this does not depend on the threshold value I specify.
I assume I am doing some mistakes with threshold values, but I do not know what is this mistake.
Could you give some suggestions or provide further details on how the image thresholding works?

Thank you in advance!
